samples/open-telemetry/application-insights/Core_8/Endpoint/MetricsShim/EmitNServiceBusMetrics.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Reflection;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Features;
+using NServiceBus.Settings;
+
+#region metrics-shim
+class EmitNServiceBusMetrics : Feature
+{
+    public EmitNServiceBusMetrics()
+    {
+        EnableByDefault();
```

Comment:
What NSB configurations might this Feature be incompatible with?  Send-only endpoints?  Others?

Reply:
I don't think there's anything incompatible; even though enabled on a send-only endpoint, it just wouldn't have any effect because no messages are consumed there.
The `OnReceivePipelineCompleted` would never be hit, and you'd never see retries, critical time, or processing time. There's also no reason to enable the shim on a SendOnly Endpoint, as it would give no added value.

Reply:
Thanks, Laila!  I am mostly asking because we have our own simple common NuGet packages that wrap up things like this we distribute out to all of our microservice teams and if they include the NuGet package in a SendOnly endpoint and the feature is set to `EnableByDefault()`, there's not really a decision point to say, don't turn it on in this context.  But that is all fine if it gets enabled and has no effect that is not an issue.  

More likely, we will change our copy of this code to have the feature off by default and enabled explicitly by each service.  But when a team says its not working, then the first question will be, "Did you turn it on?"  :-)

Reply:
Good catch @bbrandt I have updated the shim to handle send only endpoints now for completeness